coderd/database/migrations/000051_template_acl.up.sql

```diff
@@ -0,0 +1,12 @@
+BEGIN;
+
+ALTER TABLE templates ADD COLUMN user_acl jsonb NOT NULL default '{}';
```

Comment:
Why store this as a massive `jsonb` blob instead of a table? How do we plan on efficiently getting a list of all templates the user has access to?

Reply:
It's basically the same way everything else is done. You get a list of all the templates and then you filter it through the auth filter for the subset of templates that you have access to. Just wrote a test for this [here](https://github.com/coder/coder/pull/4125/commits/fe2af91eaba95f8a7fe09865cdaf722cc89e9dbc)

Reply:
For workspaces, it _kinda_ does this, but we still query by the owner so it's lessened significantly.

I guess the use-case I have in mind is 10 templates with 10 users on each... for every HTTP request we'll load 100 entries into memory and check against them? That sounds like a lot of excess when we could (I don't see why not, but maybe there's a reason) have a table that just has the user ID indexed...

Reply:
The memory overhead is unfortunate but this way plays better with rego is the short answer. We can look into rewriting it if you think it's a showstopper, I'm not sold on `jsonb` but joining tables doesn't play well with sqlc either so the maintainability of the code might suffer as a result of trying to glue everything together.